### PR TITLE
Fixes utfgrid support

### PR DESCRIPTION
**Pull request by brianreavis**
This pull request fixes utfgrid support that's been broken in 4.x (but not caught because of [a typo in render_grid.test.js](https://github.com/mapnik/node-mapnik/blob/master/test/render_grid.test.js#L16)). With the current (incomplete?) implementation, the strings in the `grid` array contain alternating NUL characters. Also, when code points fall outside of ASCII range, the encoding to UTF-8 isn't valid (due to the code points being written directly without being converted to variable-length code units).

Rather than reinvent the wheel for code point conversion / writing, I pulled in [utf8.h](https://github.com/sheredom/utf8.h).

### CODE REVIEW

**Comment by springmeyer** [comment]
Wow, thanks for fixing this @brianreavis - any idea what commit regressed?

/cc @artemp to evaluate/review.

**Comment by brianreavis** [comment]
It seems like `grid2utf()` has treated `lines` as arrays of codepoints vs arrays of utf-8 strings (code units) forever. Here it is in 3.x: https://github.com/mapnik/node-mapnik/blob/3.4.x/src/js_grid_utils.hpp#L25

In the NAN days (3.x), a line's conversion to `v8::String` got interpreted as UTF-16 due to the `uint16_t` type. It's not _really_ UTF-16, but it was close enough that it didn't cause any noticeable problems (UTF-16 characters are at least two bytes and don't get above two bytes unless you go to extremes).

When converting to NAPI, it seem that some spots got treated as UTF-16 like [here](https://github.com/mapnik/node-mapnik/commit/cba828926fce3a681db50025df0fe91d1eb935d0#diff-aa2ed5e06a175bd9a9fdf7c1a4dad9efR453), while others got treated as UTF-8 like [here](https://github.com/mapnik/node-mapnik/blob/012125683f9cbd04384208192bcf31752beac58f/src/mapnik_grid.cpp#L98). 

TL;DR: Ultimately I think the original implementation wasn't quite right (but worked well-enough) and the NAPI porting brought it to the surface (but unfortunately wasn't caught due to the test typo that also appears to have been in place for a long time https://github.com/mapnik/node-mapnik/commit/2eef229931b0b76b7d97f2ee7ad829c28a05acb9). A bit of a perfect storm of a variety of different things :)

P.S. Huge thanks for the N-API port @artemp!

**Comment by springmeyer** [comment]
Thanks @brianreavis - appreciate the thoughtfulness on this. Other question I have is what your usecase for utfgrids are and whether you might be able to replicate it with vector tiles? At Mapbox at least we were able to, so the utfgrid support here in node-mapnik is no longer used for mapbox.com related APIs.

**Comment by brianreavis** [comment]
Vector-everywhere is definitely the goal, but we're keeping raster tiles around until we perfect the vector approach (particularly information density and layout quality with reasonable performance). Our rasters are slow to build but the quality from mapnik is _super_. Hard to give it up! Not sure how long we'll be holding on to our rasters, but are for the time being.

**Comment by springmeyer** [comment]
That makes sense! Thanks for sharing.

**Comment by artemp** [comment]
thanks @brianreavis !
